### PR TITLE
Add targeting to `PresentedOfferingContext`

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation

Get more context with targeting information 

### Description

- Parse targeting information into `Offerings`
- Add targeting context into `Offerings.current`
- Add targeting context into `getCurrentOffering(forPlacement: String)`
- Post targeting context in post receipts

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -602,6 +602,7 @@ platform :ios do
 
   desc "Run BackendIntegrationTests"
   lane :backend_integration_tests do |options|
+    fetch_snapshots
```
Comment (tonidero) [inline fastlane/Fastfile:605]: Hmm is this needed? The snapshot tests for paywalls should be independent of these test I thought?

**Review by tonidero — APPROVED** [review]
Left some comments but LGTM!

File: Tests/TestingApps/PaywallsTester/PaywallsTester/Views/UpsellView.swift [hunk 2]
```diff
@@ -20,30 +20,30 @@ struct UpsellView: View {
...
+//            onRestoreCompleted: { _ in
+//                print("Restore completed")
+//            },
+//            onRestoreStarted: {
+//                print("Restore started")
+//            },
+//            onRestoreFailure: {
+//                print("Restore failed")
+//            }
+//        )
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/Views/UpsellView.swift:46]: OOOOPS... iIshouldn't have committed that lol 🤦‍♂️

File: fastlane/Fastfile (hunk 1 above)
Comment (joshdholtz) [reply fastlane/Fastfile:605]: The CI tests keep failing on UI Snapshots tests are missing in the backend tests so this was my test to see if this fixed it 🤷‍♂️ Will remove it stops the flakey

File: Sources/Purchasing/Offerings.swift [hunk 3]
```diff
@@ -41,6 +41,16 @@ import Foundation
         }
     }
 
+    internal final class Targeting: NSObject {
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply Sources/Purchasing/Offerings.swift:44]: It probably should 😅  Not sure why I did the extra work of making this an `NSObject`...

**Pull request merged**